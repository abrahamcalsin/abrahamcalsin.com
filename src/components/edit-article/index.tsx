import { TbBrandGithub } from 'react-icons/tb'
import { Flex, Icon, Link, Text } from '@chakra-ui/react'

interface EditArticleProps {
  slug: string
}

export function EditArticle(props: EditArticleProps) {
  const { slug } = props

  return (
    <Flex alignItems="center" columnGap="8px" textColor="inherit">
      <Link
        href={`https://github.com/abrahamcalsin/abrahamcalsin.com/edit/main/data/blog/${slug}.mdx`}
        rel="nofollow noopener noreferrer"
        display="flex"
        alignItems="center"
        columnGap="6px"
        fontSize="sm"
        fontStyle="normal"
        mt={{ base: '8px', sm: '0' }}
        _hover={{
          textDecoration: 'none',
        }}
        className="link-underline"
        isExternal
      >
        <Text as="span">Editar art√≠culo</Text>
        <Text as="span" w="4" display="inline-flex">
          <Icon width="full" height="full" as={TbBrandGithub} />
        </Text>
      </Link>
    </Flex>
  )
}
