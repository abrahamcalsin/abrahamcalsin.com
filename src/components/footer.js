class Footer extends HTMLElement {
  constructor() {
    super()
  }

  connectedCallback() {
    this.innerHTML = `
<div class="footer">
    <div class="footer-content">
      <div class="autor-copyright">
        <p class="autor-abraham-calsin">Â©ABRAHAM CALSIN</p>
        <span class="frontend-developer">Frontend Developer</span>
      </div>
      <!-- LINE - AUTOR AND SOCIAL -->
      <div class="divide-autor-social"><div></div></div>
      <!-- BTN IR ARRIBA -->
      <a href="javascript:void(0);" id="scroll" style="display: none"
        ><img src="Recursos/Icons/arrow-up-icon.svg" alt=""
      /></a>
      <!-- SOCIAL MEDIA -->
      <div class="social-media">
        <a
          href="https://github.com/abrahamcalsin"
          target="_blank"
          rel="noreferrer"
          ><img src="Recursos/Social-media/fa-github.svg" alt=""
        /></a>
        <a
          href="https://twitter.com/abraham_calsin"
          target="_blank"
          rel="noreferrer"
          ><img src="Recursos/Social-media/fa-twitter.svg" alt=""
        /></a>
        <a
          href="https://www.instagram.com/abrahamcalsin"
          target="_blank"
          rel="noreferrer"
          ><img src="Recursos/Social-media/fa-instagram.svg" alt=""
        /></a>
        <a
          href="https://www.facebook.com/abrahamcalsin"
          target="_blank"
          rel="noreferrer"
          ><img src="Recursos/Social-media/fa-facebook.svg" alt=""
        /></a>
        <a
          href="https://www.linkedin.com/in/abrahamcalsin"
          target="_blank"
          rel="noreferrer"
          ><img src="Recursos/Social-media/fa-linkedin.svg" alt=""
        /></a>
      </div>
    </div>
  </div>
        `
  }
}

customElements.define('footer-component', Footer)
