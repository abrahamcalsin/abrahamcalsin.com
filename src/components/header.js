class Header extends HTMLElement {
  constructor() {
    super()
  }

  connectedCallback() {
    const homeHref = this.getAttribute('brand-href')

    this.innerHTML = `
    <div class="header">
          <div class="container header-container">
            <a href="${homeHref}" class="logotype-abrahamcalsin"
              ><img
                src="Recursos/Img - Logotype/logotype-vector.svg"
                alt=""
                class="img-lo-ac"
            /></a>
            <div class="menu_navbar">
              <input type="checkbox" id="btn_menu" />
              <label for="btn_menu" class="btn_menu"
                ><i class="fa fa-bars active"></i
              ></label>
              <nav class="navbar">
                <ul>
                  <li>
                    <a
                      href="index.html"
                      class="navbar-inicio"
                      style="color: #ff7f2aff"
                      >Inicio</a
                    >
                  </li>
                  <li>
                    <a href="Pages/motivation.html" class="navbar-motivacion"
                      >motivaci√≥n</a
                    >
                  </li>
                  <li><a href="#" class="sobre-mi">sobre mi</a></li>
                  <li class="contact-abrahamcalsin"><a href="#">contacto</a></li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
    `
  }
}

customElements.define('header-component', Header)
