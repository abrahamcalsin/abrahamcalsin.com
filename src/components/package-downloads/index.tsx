import * as React from 'react'
import { FiDownload } from 'react-icons/fi'
import { Icon, Text, Tooltip } from '@chakra-ui/react'

import { getAllPackageDownloads } from '~/services/get-all-package-downloads/get-all-package-downloads'
import { numberFormat } from '~/utils/number-format'

interface PackageDownloadsProps {
  visible?: boolean
  packageName: string
}

export function PackageDownloads(props: PackageDownloadsProps) {
  const { visible, packageName } = props

  const [data, setData] = React.useState<{
    downloads: any
  } | null>(null)

  React.useEffect(() => {
    getAllPackageDownloads(packageName).then(data => {
      setData(data)
    })
  }, [])

  if (!data) return null

  const format = numberFormat(data.downloads)

  return (
    <>
      {visible && (
        <Tooltip
          label="Descargas del paquete en Npm"
          placement="top"
          aria-label="Descargas del paquete en Npm"
          py="1"
          px="1.5"
          gutter={10}
          bg="primary.700"
          _dark={{
            backgroundColor: 'primary.100',
          }}
        >
          <Text
            as="span"
            display="inline-flex"
            alignItems="center"
            ml="2"
            gap="1.5"
            fontSize="md"
            fontWeight="medium"
            textColor="primary.400"
            _dark={{
              textColor: 'secondary.400',
            }}
          >
            <Icon w="4" height="full" as={FiDownload} />
            {format}
          </Text>
        </Tooltip>
      )}
    </>
  )
}
