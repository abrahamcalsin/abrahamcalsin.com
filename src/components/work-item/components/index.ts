export * from './work-item-icon'
