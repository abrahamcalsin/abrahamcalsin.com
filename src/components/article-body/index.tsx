import * as React from 'react'
import { CgCoffee } from 'react-icons/cg'
import { Box, Divider, Flex, Heading, Icon, Text } from '@chakra-ui/react'

import { DateFormat } from '~/components/date-format'
import { EditArticle } from '~/components/edit-article'
import { MdxComponents } from '~/components/mdx-components/mdx-components'
import { ShareArticle } from '~/components/share-article'

interface ArticleBodyProps {
  title: string
  date: string
  content: string
  articleSlug: string
  shareTitle: string
  readingTime: string
}

export function ArticleBody(props: ArticleBodyProps) {
  const { title, date, content, articleSlug, shareTitle, readingTime } = props

  return (
    <Box my={{ base: '0', sm: '16' }}>
      <div>
        <Heading
          as="h1"
          fontWeight="black"
          textColor="primary.800"
          fontSize={{ base: '3xl', sm: '4xl' }}
          _dark={{
            textColor: 'primary.200',
          }}
        >
          {title}
        </Heading>
        <Flex
          justifyContent="space-between"
          textColor="primary.600"
          fontSize={{ base: 'sm', sm: 'md' }}
          mt={{ base: '4', sm: '7' }}
          mb="4"
          _dark={{
            textColor: 'primary.400',
          }}
        >
          <div>
            <DateFormat date={date} locale="es" />
          </div>
          <Box display={{ base: 'none', sm: 'flex' }} alignItems="center" gap="2">
            <Icon w="4" height="full" as={CgCoffee} />
            <span>{readingTime}</span>
          </Box>
        </Flex>
      </div>
      <Box
        w="full"
        maxWidth="none"
        textColor="primary.600"
        mt={{ base: '5', sm: '7' }}
        _dark={{
          textColor: 'primary.400',
        }}
      >
        <MdxComponents rawContent={content} />
      </Box>
      <Divider my="7" borderColor="primary.400" _dark={{ borderColor: 'primary.600' }} />
      <Flex
        alignItems="center"
        columnGap="2"
        textColor="primary.600"
        _dark={{
          textColor: 'primary.400',
        }}
      >
        <ShareArticle articleLink={articleSlug} articleTitle={shareTitle} />
        <Text as="span">â€¢</Text>
        <EditArticle slug={articleSlug} />
      </Flex>
    </Box>
  )
}
