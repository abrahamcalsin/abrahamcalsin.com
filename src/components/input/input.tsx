import * as React from 'react'
import { Input as ChakraInput, InputGroup, InputLeftElement, InputRightElement } from '@chakra-ui/react'

import { InputProps } from '~/typings/components/input'

export function Input(props: InputProps) {
  const { type, variant, placeholder, value, leftIcon, rightIcon, handleChange } = props

  return (
    <InputGroup>
      {leftIcon ? <InputLeftElement pointerEvents="none">{leftIcon}</InputLeftElement> : null}
      <ChakraInput
        type={type}
        role="input"
        variant={variant}
        placeholder={placeholder}
        value={value}
        onChange={handleChange}
      />
      {rightIcon ? <InputRightElement pointerEvents="none">{rightIcon}</InputRightElement> : null}
    </InputGroup>
  )
}
