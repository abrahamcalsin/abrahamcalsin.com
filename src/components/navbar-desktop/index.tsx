import * as React from 'react'
import { Container, Flex, Link } from '@chakra-ui/react'
import clsx from 'clsx'
import RouterLink from 'next/link'
import { useRouter } from 'next/router'

import { DarkThemeButton } from '~/components/dark-theme-button'
import { LogoAbrahamCalsin } from '~/components/svgs'

interface NavigationLinkProps {
  href: string
  textLink: string
}

function NavigationLink(props: NavigationLinkProps) {
  const { href, textLink } = props

  const router = useRouter()

  const isActive = router.pathname === href

  return (
    <RouterLink href={href} passHref>
      <Link
        _hover={{
          textColor: 'secondary.400',
        }}
        pointerEvents={isActive ? 'none' : undefined}
        textColor={clsx(isActive && 'secondary.400')}
      >
        {textLink}
      </Link>
    </RouterLink>
  )
}

export default function NavbarDesktop() {
  return (
    <Container maxW="52rem" display={{ base: 'none', sm: 'flex' }} justifyContent="space-between" alignItems="center">
      <RouterLink href="/" passHref>
        <Link width="20">
          <LogoAbrahamCalsin />
        </Link>
      </RouterLink>
      <Flex alignItems="center" fontSize="lg" fontWeight="bold" columnGap="5">
        <NavigationLink href="/" textLink="Inicio" />
        <NavigationLink href="/about-me" textLink="Sobre Mi" />
        <NavigationLink href="/projects" textLink="Proyectos" />
        <NavigationLink href="/blog" textLink="Blog" />
        <DarkThemeButton />
      </Flex>
    </Container>
  )
}
