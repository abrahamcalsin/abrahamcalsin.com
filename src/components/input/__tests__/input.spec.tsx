import * as React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Input } from '~/components/input'
import type { InputProps } from '~/typings/components/input'

const mockHandleChange = jest.fn()

const setup = (props?: InputProps) => {
  return render(<Input {...props} />)
}

describe('Input', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders the default input', () => {
    const view = setup()

    expect(view.baseElement).toMatchSnapshot()
  })

  it('renders an input with left element', () => {
    const view = setup({
      leftIcon: <span data-testid="left-element" />,
    })

    const leftElement = screen.getByTestId('left-element')

    expect(leftElement).toBeInTheDocument()

    expect(view.baseElement).toMatchSnapshot()
  })

  it('renders an input with right element', () => {
    const view = setup({
      rightIcon: <span data-testid="right-element" />,
    })

    const rightElement = screen.getByTestId('right-element')

    expect(rightElement).toBeInTheDocument()

    expect(view.baseElement).toMatchSnapshot()
  })

  it('renders an input with left and right element', () => {
    const view = setup({
      leftIcon: <span data-testid="left-element" />,
      rightIcon: <span data-testid="right-element" />,
    })

    const leftElement = screen.getByTestId('left-element')

    expect(leftElement).toBeInTheDocument()

    const rightElement = screen.getByTestId('right-element')

    expect(rightElement).toBeInTheDocument()

    expect(view.baseElement).toMatchSnapshot()
  })

  it('writes in input', async () => {
    const view = setup({
      handleChange: mockHandleChange,
    })

    const input = screen.getByRole('input')

    await userEvent.type(input, 'typing')

    expect(input).toHaveValue('typing')

    expect(view.baseElement).toMatchSnapshot()
  })
})
