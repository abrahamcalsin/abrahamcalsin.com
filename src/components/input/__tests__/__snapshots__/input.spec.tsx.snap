// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders an input with left and right element 1`] = `
<body>
  <div>
    <div
      class="chakra-input__group css-4302v8"
    >
      <div
        class="chakra-input__left-element css-1u2k0bl"
      >
        <span
          data-testid="left-element"
        />
      </div>
      <input
        class="chakra-input css-0"
        role="input"
        value=""
      />
      <div
        class="chakra-input__right-element css-1hx2jp8"
      >
        <span
          data-testid="right-element"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Input renders an input with left element 1`] = `
<body>
  <div>
    <div
      class="chakra-input__group css-4302v8"
    >
      <div
        class="chakra-input__left-element css-1u2k0bl"
      >
        <span
          data-testid="left-element"
        />
      </div>
      <input
        class="chakra-input css-0"
        role="input"
        value=""
      />
    </div>
  </div>
</body>
`;

exports[`Input renders an input with right element 1`] = `
<body>
  <div>
    <div
      class="chakra-input__group css-4302v8"
    >
      <input
        class="chakra-input css-0"
        role="input"
        value=""
      />
      <div
        class="chakra-input__right-element css-1hx2jp8"
      >
        <span
          data-testid="right-element"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Input renders the default input 1`] = `
<body>
  <div>
    <div
      class="chakra-input__group css-4302v8"
    >
      <input
        class="chakra-input css-0"
        role="input"
        value=""
      />
    </div>
  </div>
</body>
`;

exports[`Input writes in input 1`] = `
<body>
  <div>
    <div
      class="chakra-input__group css-4302v8"
    >
      <input
        class="chakra-input css-0"
        role="input"
        value=""
      />
    </div>
  </div>
</body>
`;
